STX.REST.RESTFulSense.Clients/Services/Foundations/HttpExchanges/HttpExchangeService.Validations.cs

```diff
@@ -0,0 +1,204 @@
+
+            ValidateHttpExchange(
+                httpExchange,
+                defaultHttpMethod,
+                defaultHttpVersion,
+                defaultHttpVersionPolicy);
+        }
+
+        private static void ValidateHttpExchange(
+            HttpExchange httpExchange,
+            HttpMethod httpMethod,
+            Version httpVersion,
+            HttpVersionPolicy httpVersionPolicy)
+        {
+            ValidateHttpExchangeRequest(
+                httpExchangeRequest: httpExchange.Request,
+                httpMethod,
+                httpVersion,
+                httpVersionPolicy);
+        }
+
+        private static void ValidateHttpExchangeRequest(
+            HttpExchangeRequest httpExchangeRequest,
+            HttpMethod httpMethod,
+            Version httpVersion,
+            HttpVersionPolicy httpVersionPolicy)
+        {
+            ValidateHttpExchangeRequestNotNull(httpExchangeRequest);
+
+            ValidateInputArgument(
+                (Rule: IsInvalid(httpExchangeRequest.BaseAddress),
+                    Parameter: nameof(HttpExchangeRequest.BaseAddress)),
+
+                (Rule: IsInvalid(httpExchangeRequest.RelativeUrl),
+                    Parameter: nameof(HttpExchangeRequest.RelativeUrl)),
+
+                (Rule: IsInvalidHttpMethod(httpExchangeRequest.HttpMethod, httpMethod),
+                    Parameter: nameof(HttpExchangeRequest.HttpMethod)),
+
+                (Rule: IsInvalidHttpVersion(httpExchangeRequest.Version, httpVersion),
+                    Parameter: nameof(HttpExchangeRequest.Version)),
+
+                (Rule: IsInvalidHttpVersionPolicy(httpExchangeRequest.VersionPolicy, httpVersionPolicy),
+                    Parameter: nameof(HttpExchangeRequest.VersionPolicy)));
+
+            ValidateHttpExchangeRequestHeaders(httpExchangeRequestHeaders: httpExchangeRequest.Headers);
+        }
+
+        private static void ValidateHttpMethod(HttpExchangeRequest httpExchangeRequest, HttpMethod httpMethod)
+        {
+            ValidateInputArgument((
+                Rule: IsInvalidHttpMethod(httpExchangeRequest.HttpMethod, httpMethod),
+                Parameter: nameof(HttpExchangeRequest.HttpMethod)));
+        }
+
+        private static void ValidateVersion(HttpExchangeRequest httpExchangeRequest, Version httpVersion)
+        {
+            ValidateInputArgument((
+                Rule: IsInvalidHttpVersion(httpExchangeRequest.Version, httpVersion),
+                Parameter: nameof(HttpExchangeRequest.Version)));
+        }
+
+        private static void ValidateHttpExchangeNotNull(HttpExchange httpExchange)
+        {
+            if (httpExchange is null)
+            {
+                throw new NullHttpExchangeException(
+                    message: "Null HttpExchange error occurred, fix errors and try again.");
+            }
+        }
+
+        private static void ValidateHttpExchangeRequestNotNull(HttpExchangeRequest httpExchangeRequest)
```

Comment:
Not used

Reply:
FYI, line 53 needs removed which causes build error and tests break.

Reply:
@cjdutoit @glhays  I wanna confirm if you need this isn't needed. because it's been used. line 53 is the only place we are validating if HttpExchangeRequest is not null.

Reply:
My fault, I think this was meant for line 88.

My issue on this one is the ValidateInputArgument.

Do this instead:
```cs
            if (httpExchange is null)
            {
                throw new NullHttpExchangeRequestException(
                    message: "Null HttpExchangeRequest error occurred, fix errors and try again.");
            }
```